### ISSUE TITLE
Dynamic weight change from in game factors

### ISSUE THREAD

**Issue by Ultimushadow** [AnOpenSauceDev/Weight#4]
Hey there, I was wondering if this was possible with the given arrangement, and if not to see if it's possible to add.

One of the armor sets in my pack is a full set of Turtle armor. It provides an alright amount of defense, so I want it to be slower on land, but I was thinking that in the water, the set would stop slowing the player.

That got me thinking of other ways to induce a dynamic change like this, for instance, Netherite could reduce the speed penalty when the player is on fire, which uses a tag. You could have armor which is more effective in a certain dimension, like nullifying the speed penalty of Gold in the Nether to incentivize using it further.
Some other ideas don't quite fit the vanilla sets, but I can imagine armor which would be more/less impactful when worn in different biomes, or at different times of day, altitudes, light levels, etc. If you included potion effects, there could even be a potion effect designed to temporarily shoulder some of the burden, or maybe an enchantment for a more permanent offset to work towards. 

I believe this can all be done theoretically by datapacks and messing with speed values but that would be a monster of a resource drain for the server so I figure it might be possible for you to add the ability for the fields I just mentioned and the corresponding weight change for those checks. Let me know what you think!

**Comment by AnOpenSauceDev** [comment]
Definitely possible using the raw Java API, but a user-friendly alternative would be a good idea.

**Comment by AnOpenSauceDev** [comment]
Almost a year late to the party, but I got back into working on Armour Weight again, and I've implemented this functionality as a datapack function that gets automatically called when wearing a piece of armour, and can be used alongside the already existing `CalculateWeight` command and scoreboard to do just about anything you want.

 This way it shouldn't have much of an impact on performance (just check if weight > value, modify entity attribute). I always intended to keep Armour Weight's functionality pretty lightweight, but this is a good compromise in my opinion.
